File: src/main/java/no/nav/fo/veilarbregistrering/db/DatabaseMigrator.java
package no.nav.fo.veilarbregistrering.db;

import org.flywaydb.core.Flyway;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.annotation.Configuration;

import javax.annotation.PostConstruct;
import javax.sql.DataSource;


@Configuration
public class DatabaseMigrator {

    private final DataSource dataSource;

    @Autowired
    public DatabaseMigrator(DataSource dataSource) {
        this.dataSource = dataSource;
    }

    @PostConstruct
    public void migrateDb() {
        Flyway flyway = new Flyway();
        flyway.setDataSource(dataSource);
        flyway.migrate();
    }

}

File: src/main/java/no/nav/fo/veilarbregistrering/db/DatabaseMigrator.kt
package no.nav.fo.veilarbregistrering.db

import org.flywaydb.core.Flyway
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.context.annotation.Configuration
import javax.annotation.PostConstruct
import javax.sql.DataSource

@Configuration
class DatabaseMigrator @Autowired constructor(private val dataSource: DataSource) {
    @PostConstruct
    fun migrateDb() {
        val flyway = Flyway()
        flyway.dataSource = dataSource
        flyway.migrate()
    }
}